### PR TITLE
[ALLUXIO-3256] Reduce mem usage of ACLs

### PR DESCRIPTION
**Pull request by gpang**
https://alluxio.atlassian.net/browse/ALLUXIO-3256

### CODE REVIEW

**Comment by gpang** [comment]
@yuzhu

File: core/common/src/main/java/alluxio/security/authorization/AccessControlList.java [hunk 1]
```diff
@@ -71,17 +69,14 @@
   public static final String OWNING_GROUP_FIELD = "owningGroup";
   public static final String STRING_ENTRY_FIELD = "stringEntries";
 
-  /** Key representing owning user in {@link #mUserActions}. */
+  /** Keys representing owning user and group for proto ser/de. */
```
Comment (yuzhu) [inline core/common/src/main/java/alluxio/security/authorization/AccessControlList.java]: this comment should apply to the constants above `OWNER_FIELD` , `OWNING_GROUP_FIELD` , `STRING_ENTRY_FIELD`

File: core/common/src/main/java/alluxio/security/authorization/DefaultAccessControlList.java [hunk 2]
```diff
@@ -189,18 +187,18 @@ public boolean equals(Object o) {
       return false;
     }
     DefaultAccessControlList that = (DefaultAccessControlList) o;
+    boolean extendedEquals = (mExtendedEntries == null && that.mExtendedEntries == null)
```
Comment (yuzhu) [inline core/common/src/main/java/alluxio/security/authorization/DefaultAccessControlList.java]: should this be changed to consider `null` and empty `extendedEntries` the same?  like `AccessControlList`?

File: core/common/src/main/java/alluxio/security/authorization/AccessControlList.java (hunk 1 above)
Comment (gpang) [reply core/common/src/main/java/alluxio/security/authorization/AccessControlList.java]: Moved the comment.

File: core/common/src/main/java/alluxio/security/authorization/DefaultAccessControlList.java (hunk 2 above)
Comment (gpang) [reply core/common/src/main/java/alluxio/security/authorization/DefaultAccessControlList.java]: I updated this (like acl).

**Pull request merged**